ValidateToken function Fix

tanushreegorai opened the pull request:
…g with the env var

### :hammer_and_wrench: Description
`validateGitPatToken()` now takes `tokenFromProvider` as a parameter. It takes the PAT token from the provider first if the user has passed it from there, else it looks up in the environment variable.
<!-- What code changed, and why? -->
<!-- Share any information that will help a reviewer without any prior knowledge to build context on why this change was made -->

### :link: External Links

<!-- Jira task (add issue number and uncomment below) -->
<!-- https://hashicorp.atlassian.net/browse/<ISSUE-NUMBER> -->

<!-- Related RFCs, PRs, Slack threads -->

### :+1: Definition of Done

<!-- Check off any testing that has been done. If no testing has been done yet, be sure to let the reviewer know what future testing is planned -->

- [x] Unit tests added?

<!-- Please uncomment the checkbox below if a database query changed -->
<!-- - [ ] Integration tests added? -->

<!-- Please uncomment checkbox below if e2e tests changed -->
<!-- - [ ] E2E tests run? -->
<!-- If e2e tests were run in integration, include a link to e2e-test GHA run -->
<!-- If the e2e test were run locally, paste the output below -->

<!-- Include any additional details and screenshots to help the reviewer understand what has been tested -->

sujaysamanta committed:
:bug: fix(git_token): Remove git toekn validation as it is optional

tanushreegorai committed:
Passed PAT token from the provider to the validateToken function along with the env var

dependabot[bot] committed:
build(deps): bump golang.org/x/oauth2 from 0.27.0 to 0.28.0 (#103)

a-anurag27 commented on internal/util/vcs/git/gitUtil.go:
```diff
@@ -229,12 +229,20 @@ func (g *gitUtil) OpenRepository(repoPath string) (*git.Repository, error) {
 	}
 
-	gitPatToken, isSet := os.LookupEnv("TF_GIT_PAT_TOKEN")
+	var gitPatToken string
```
Please see if this can be refactored, isSet seems redundant.
```suggestion
	gitPatToken := tokenFromProvider
	if gitPatToken == "" {
		if token, exists := os.LookupEnv("TF_GIT_PAT_TOKEN"); exists {
			gitPatToken = token
		} else {
			return "", cliErrs.ErrTfGitPatTokenNotSet
		}
	}
```

tanushreegorai replied:
Can you please elaborate, how isSet is redundant? Here it is being set to true if we are getting the token from the provider (it doesn't depend on the env variable  in this case) and second case is that it will look up in the env var if the git pat token is set or not and take the value accordingly.

a-anurag27 commented on internal/util/vcs/git/remote_svc_provider/github_svc_provider.go:
```diff
@@ -28,8 +28,8 @@ type GithubSvcProvider interface {
-	if _, err := g.git.GetGitToken(g.git.GetRemoteServiceProvider(repoUrl)); err != nil {
+func (g *githubSvcProvider) ValidateToken(repoUrl string, repoIdentifier string, tokenFromProvider string) (string, error) {
+	if _, err := g.git.GetGitToken(g.git.GetRemoteServiceProvider(repoUrl), tokenFromProvider); err != nil {
 		suggestions, gitTokenErr := gitTokenErrorHandler(err)
```
Assigned variables can be removed.
```suggestion
		return gitTokenErrorHandler(err)
```

a-anurag27 commented on internal/util/vcs/git/remote_svc_provider/gitlab_svc_provider.go:
```diff
@@ -26,8 +26,8 @@ type GitlabSvcProvider interface {
-	if _, err := g.git.GetGitToken(g.git.GetRemoteServiceProvider(repoUrl)); err != nil {
+func (g *gitlabSvcProvider) ValidateToken(repoUrl string, projectIdentifier string, tokenFromProvider string) (string, error) {
+	if _, err := g.git.GetGitToken(g.git.GetRemoteServiceProvider(repoUrl), tokenFromProvider); err != nil {
 		suggestions, gitTokenErr := gitTokenErrorHandler(err)
```
Same comment on variable assignment.

a-anurag27 reviewed (changes requested):
Minor suggestions.

tanushreegorai committed:
PR Review

tanushreegorai committed:
Linter fix

dependabot[bot] committed:
build(deps): bump golang.org/x/oauth2 from 0.28.0 to 0.29.0 (#110)

Bumps [golang.org/x/oauth2](https://github.com/golang/oauth2) from 0.28.0 to 0.29.0.
- [Commits](https://github.com/golang/oauth2/compare/v0.28.0...v0.29.0)

---
updated-dependencies:
- dependency-name: golang.org/x/oauth2
  dependency-version: 0.29.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump gitlab.com/gitlab-org/api/client-go (#109)

Bumps [gitlab.com/gitlab-org/api/client-go](https://gitlab.com/gitlab-org/api/client-go) from 0.124.0 to 0.127.0.
- [Release notes](https://gitlab.com/gitlab-org/api/client-go/tags)
- [Commits](https://gitlab.com/gitlab-org/api/client-go/compare/v0.124.0...v0.127.0)

---
updated-dependencies:
- dependency-name: gitlab.com/gitlab-org/api/client-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/hashicorp/terraform-plugin-testing (#107)

dependabot[bot] committed:
build(deps): bump github.com/hashicorp/go-tfe from 1.76.0 to 1.78.0 (#115)

dependabot[bot] committed:
build(deps): bump github.com/go-git/go-git/v5 from 5.14.0 to 5.16.0 (#113)

dependabot[bot] committed:
build(deps): bump github.com/hashicorp/terraform-exec (#111)

a-anurag27 reviewed (dismissed):
lgtm, rebase before merge.

tanushreegorai committed:
removed gitPatToken validation if gitPatToken is not configured in provider

sujaysamanta committed:
Merge branch 'feature/v1.1.0' into validatePatTokenFix
